docs/image-building.md

```diff
@@ -0,0 +1,175 @@
+[startup-script]: https://cloud.google.com/compute/docs/instances/startup-scripts
+
+## Toolkit-supported approaches
+
+Toolkit supports Packer modules as distinct deployment groups each containing
+only 1 Packer module that must be identified using `kind`. For example:
+
+```yaml
+- group: packer
+  modules:
+  - id: custom-image
+    source: modules/packer/custom-image
+    kind: packer
+    use: []
+    settings: {}
+```
+
+### Toolkit Packer module
+
+The Toolkit includes a [Packer module](../modules/packer/custom-image/README.md)
+for building custom images in blueprints. This module has been designed with
+requirements and best practices in mind. For example, it works effectively with
+startup scripts and, by default, will enable IAP tunnels for provisioners that
+need them.
+
+#### Windows support
+
+The Toolkit Packer module provides limited support for building custom VM images
+based on the [Windows images][windows-images] published by Google. The custom VM
+images can be used in blueprints so long as the underlying scheduler and
+workload supports Windows. Windows solutions do not receive the same level of
+testing as Linux solutions so you should anticipate that there will not be
+functionality parity. Please file [issues] when encountering specific problems
+and [feature requests][features] when requesting new functionality.
+
+[windows-images]: https://cloud.google.com/compute/docs/images/os-details#windows_server
+[issues]: https://github.com/GoogleCloudPlatform/hpc-toolkit/issues
+[features]: https://github.com/GoogleCloudPlatform/hpc-toolkit/discussions/categories/ideas-and-feature-requests
+
+### External Packer modules
+
+The Toolkit supports Packer modules developed by 3rd parties -- including ones
+that you have developed! -- hosted via git or GitHub. We recommend reading the
+module documentation on:
+
+- [GitHub-hosted modules and packages](https://github.com/tpdownes/hpc-toolkit/blob/packer_documentation/modules/README.md#github-hosted-modules-and-packages)
+- [GitHub-hosted Packer modules](https://github.com/tpdownes/hpc-toolkit/blob/packer_documentation/modules/README.md#github-hosted-packer-modules)
+
+In particular, the Toolkit recommends using double-slash (`//`) notation to
+identify the root of the git repository. Doing so will ensure that the Packer
+module has access to all files within the repository even if it is located in
+a subdirectory.
+
+> **_NOTE:_** If you fail to use `//` notatation when referring to the [SchedMD
+> packer module][schedmd-packer], it will fail to execute because it refers
+> to Ansible playbooks by a relative path (`../ansible`) that will not be
+> downloaded.
+
+For example, to address the issue noted above:
+
+```yaml
+deployment_groups:
+- group: primary
+  modules:
+  - id: network1
+    source: modules/network/vpc
+
+- group: packer
+  modules:
+  - id: custom-image
+    source: github.com/SchedMD/slurm-gcp//packer?ref=5.7.4&depth=1
```

Comment:
I am not familiar with the `depth` argument. What is this doing?

Reply:
Mentioned here (graf or 2 down): https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/README.md#github-hosted-modules-and-packages

TL;DR: for tags and branches (not commit hashes): you can use this and git will only fetch that many commits before checkout-ing your reference. In theory and hopefully practice: faster, less space